libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/LGActivityGenerator.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using AdaptiveCards;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.LanguageGeneration
+{
+    public class LGActivityGenerator
+    {
+        private TemplateEngine templateEngine;
+
+        public const string TextTemplateId = "TextTemplateId";
+
+        public const string AdaptiveCardTemplateId = "AdaptiveCardTemplateId";
+
+        public const string CardTemplateId = "CardTemplateId";
+
+        public const string Attachments = "Attachments";
+
+        public const string Separtor = "Separtor";
+
+        public const string AttachmentLayoutType = "AttachmentLayoutType";
+
+        public LGActivityGenerator(TemplateEngine templateEngine)
+        {
+            this.templateEngine = templateEngine;
+        }
+
+        public Activity Generate(Dictionary<string, object> options, object scope)
```

Comment:
Looks like, making this options a free Dict is not a good idea, for better clarification and for less confusion.  I think we can make this strong typed. 
Perhaps:
class ActivityGenerationConfig {
public TextTemplateId;
public TextSpeakSeperator;
public List<AttachmentGenerationConfig> Attachments;
}

public class AttachmentGenerationConfig {
  public AttachementTemplateId;
  public IsAdaptiveCard (y/n);
}

Reply:
fixed